api-client/src/main/java/com/takipi/api/client/request/alertssettings/UpdateAlertsSettingsRequest.java

```diff
@@ -0,0 +1,313 @@
+			
+			return this;
+		}
+		
+		public Builder setHipChatRoom(String hipChatRoom) {
+			this.defaultAlertsSettings.hip_chat.room = hipChatRoom;
+			
+			return this;
+		}
+		
+		public Builder setHipChatUrl(String hipChatUrl) {
+			this.defaultAlertsSettings.hip_chat.url = hipChatUrl;
+			
+			return this;
+		}
+		
+		public Builder setPagerDutyServiceIntegrationKey(String pagerDutyServiceIntegrationKey) {
+			this.defaultAlertsSettings.pager_duty.service_integration_key = pagerDutyServiceIntegrationKey;
+			
+			return this;
+		}
+		
+		public Builder setWebhookUrl(String webhookUrl) {
+			this.defaultAlertsSettings.webhook.webhook_url = webhookUrl;
+			
+			return this;
+		}
+		
+		public Builder setServiceNowUrl(String serviceNowUrl) {
+			this.defaultAlertsSettings.service_now.url = serviceNowUrl;
+			
+			return this;
+		}
+		
+		public Builder setServiceNowUserId(String serviceNowUserId) {
+			this.defaultAlertsSettings.service_now.user_id = serviceNowUserId;
+			
+			return this;
+		}
+		
+		public Builder setServiceNowPassword(String serviceNowPassword) {
+			this.defaultAlertsSettings.service_now.password = serviceNowPassword;
+			
+			return this;
+		}
+		
+		public Builder setServiceNowTable(String serviceNowTable) {
+			this.defaultAlertsSettings.service_now.table = serviceNowTable;
+			
+			return this;
+		}
+		
+		@Override
+		protected void validate() {
+			super.validate();
+			
+			if ((this.defaultAlertsSettings.slack.inhook_url != null) &&
+				(Strings.isNullOrEmpty(this.defaultAlertsSettings.slack.inhook_url)))
+			{
+				throw new IllegalArgumentException("No inhook URL supplied for Slack inhook connection request");
+			}
+			
+			if ((this.defaultAlertsSettings.hip_chat.token != null) ||
+				(this.defaultAlertsSettings.hip_chat.room != null))
+			{
+				if ((Strings.isNullOrEmpty(this.defaultAlertsSettings.hip_chat.token)) ||
+				 	(Strings.isNullOrEmpty(this.defaultAlertsSettings.hip_chat.room)))
+				{
+					throw new IllegalArgumentException("No room or token supplied for HipChat connection request");
+				}
+			}
+			
+			if ((this.defaultAlertsSettings.pager_duty.service_integration_key != null) &&
+				(Strings.isNullOrEmpty(this.defaultAlertsSettings.pager_duty.service_integration_key)))
+			{
+				throw new IllegalArgumentException("No integration key supplied for PagerDuty connection request");
+			}
+			
+			if ((this.defaultAlertsSettings.webhook.webhook_url != null) &&
+				(Strings.isNullOrEmpty(this.defaultAlertsSettings.webhook.webhook_url)))
+			{
+				throw new IllegalArgumentException("No webhook URL supplied for webhook request");
+			}
+			
+			if ((this.defaultAlertsSettings.service_now.url != null) ||
+				(this.defaultAlertsSettings.service_now.user_id != null) ||
+				(this.defaultAlertsSettings.service_now.password != null) ||
+				(this.defaultAlertsSettings.service_now.table != null))
+			{
+				if ((Strings.isNullOrEmpty(this.defaultAlertsSettings.service_now.url)) ||

```

Comment:
This check basically says "if any of those params isn't null, all of them has to be non-null-non-empty"

However, in the request `postData` you explicitly only add them one-by-one.
This tells me that the API endpoint allows you to just change one of the parameters.

If the API endpoint does allow that, then the check here is incorrect, as it prohibits something the API endpoint allows.
If the API endpoint doesn't allow that, the then building inside `postData` is doing a lot of unnecessary work.

This holds true for a lot of the settings here.